### PR TITLE
Move out globalization code from runtime

### PR DESCRIPTION
**Pull request by mkhamoyan**
Currently we compile the globalization code twice: 
1) directly into the mono runtime, by duplicating the logic
2) from src/native/System.Globalization.Native

By this PR we will move the globalization code out of the mono runtime.

### CODE REVIEW

**Comment by vargaz** [comment]
If the code in native-library.c is removed, how are these pinvoke methods going to be resolved ?

**Comment by mkhamoyan** [comment]
> If the code in native-library.c is removed, how are these pinvoke methods going to be resolved ?

https://github.com/dotnet/runtime/blob/bd5f6eb73f6dffe91386f75d3ed5531ddcb2aeb6/src/coreclr/vm/pinvokeoverride.cpp#L32 I think this part will handle it

**Comment by jkoritzinsky** [comment]
> > If the code in native-library.c is removed, how are these pinvoke methods going to be resolved ?
> 
> https://github.com/dotnet/runtime/blob/bd5f6eb73f6dffe91386f75d3ed5531ddcb2aeb6/src/coreclr/vm/pinvokeoverride.cpp#L32
> 
> I think this part will handle it

This is in the CoreCLR code-base, which isn't used by Mono at all. The code you removed in native-library.c is the equivalent code in Mono.

**Comment by mkhamoyan** [comment]
> This is in the CoreCLR code-base, which isn't used by Mono at all. The code you removed in native-library.c is the equivalent code in Mono.

I see, but it is not clear for me why after removing code from `native-library.c`, tests which pinvoke globalization code didn't complain.

**Comment by mkhamoyan** [comment]
@lewing @akoeplinger could you please review

**Review by akoeplinger — APPROVED** [review]

File: src/mono/mono.proj [hunk 1]
```diff
@@ -444,9 +433,6 @@
     <ItemGroup Condition="'$(TargetsBrowser)' == 'true'">
       <_MonoCMakeArgs Include="-DENABLE_LLVM_RUNTIME=1"/>
       <_MonoCMakeArgs Include="-DEMSCRIPTEN_SYSTEM_PROCESSOR=wasm"/>
-      <_MonoCFLAGS Condition="'$(WasmEnableThreads)' == 'true'" Include="$(EscapedQuoteW)-I$([MSBuild]::NormalizePath('$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)', 'runtimes', 'browser-wasm-threads', 'native', 'include'))$(EscapedQuoteW)"/>
-
-      <_MonoCFLAGS Condition="'$(WasmEnableThreads)' != 'true'" Include="$(EscapedQuoteW)-I$([MSBuild]::NormalizePath('$(PkgMicrosoft_NETCore_Runtime_ICU_Transport)', 'runtimes', 'browser-wasm', 'native', 'include'))$(EscapedQuoteW)"/>
```
[the remark this answers is not in the document]
Comment (mkhamoyan) [reply src/mono/mono.proj:449]: There are CI failures, reverted this part to validate if this caused.

File: src/native/libs/System.Globalization.Native/CMakeLists.txt [hunk 2]
```diff
@@ -135,8 +131,9 @@ if (NOT CLR_CMAKE_TARGET_BROWSER AND NOT CLR_CMAKE_TARGET_WASI AND NOT CLR_CMAKE
     set(NATIVEGLOBALIZATION_SOURCES ${NATIVEGLOBALIZATION_SOURCES} pal_timeZoneInfo.c)
 endif()
 
-if (MSVC)
-    set_source_files_properties(${NATIVEGLOBALIZATION_SOURCES} PROPERTIES LANGUAGE CXX)
+if (MSVC AND MSVC_VERSION LESS 1939)
```
Comment (am11) [inline src/native/libs/System.Globalization.Native/CMakeLists.txt:134]: Our minimal VS version [offsets](https://learn.microsoft.com/en-us/cpp/overview/visual-cpp-language-conformance?view=msvc-160#c-standard-library-features-1) C99 support required here, so this check is unnecessary.

https://github.com/dotnet/runtime/blob/79a1c917798a1fa27455a8683657bc2789eee939/eng/native/init-vs-env.cmd#L57

Comment (akoeplinger) [reply src/native/libs/System.Globalization.Native/CMakeLists.txt:134]: `__typeof__` support is new in MSVC 19.39: https://learn.microsoft.com/en-us/cpp/c-language/typeof-c?view=msvc-170

**Comment by mkhamoyan** [comment]
Failures are not related.

**Pull request merged**